prow/config/config.go

```diff
@@ -132,24 +132,29 @@ func Load(path string) (*Config, error) {
 
 func parseConfig(c *Config) error {
 	// Ensure that presubmit regexes are valid.
-	for _, v := range c.Presubmits {
-		if err := setRegexes(v); err != nil {
+	for _, vs := range c.Presubmits {
+		if err := setRegexes(vs); err != nil {
 			return fmt.Errorf("could not set regex: %v", err)
 		}
+		for v := range vs {
+			if vs[v].Agent == "kubernetes" && vs[v].Spec == nil {
```

Comment:
For these I think we have [kube.KubernetesAgent](https://github.com/kubernetes/test-infra/blob/7d92f76445d73a606978b5b5e5d3837207e9a65b/prow/kube/prowjob.go#L45-L48).